### PR TITLE
Changing the way users can enable logging

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/LoggerFactory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    it("creates NullLogger when logging is set to null", () => {
+        expect(LoggerFactory.createLogger(null).constructor.name).toBe("NullLogger");
+    });
+
+    it("creates ConsoleLogger when log level specified", () => {
+        expect(LoggerFactory.createLogger(LogLevel.Information).constructor.name).toBe("ConsoleLogger");
+    });
+
+    it("creates ConsoleLeogger when logging is set LogLevel", () => {
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/LoggerFactory.spec.ts]: This test name seems wrong

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { ITransport, TransferMode, TransportType, WebSocketTransport, ServerSent
 import { IHttpClient, HttpClient } from "./HttpClient"
 import { IHttpConnectionOptions } from "./IHttpConnectionOptions"
 import { ILogger, LogLevel } from "./ILogger"
-import { NullLogger } from "./Loggers"
+import { NullLogger, ConsoleLogger, LoggerFactory } from "./Loggers"
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts]: Do you need to import ConsoleLogger here?

Comment (moozzyk) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts]: probably not - I started cleaning some imports/exports but decided it did not belong in this PR and must have missed it.

File: client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js [hunk 3]
```diff
@@ -31,7 +30,7 @@ describe('hubConnection', function () {
 
             it('can stream server method and receive result', function (done) {
                 var logger = new signalR.ConsoleLogger(signalR.LogLevel.Information);
```
Comment (moozzyk) [inline client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js]: not needed/used anymore

**Comment by moozzyk** [comment]
:up::date:

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/LoggerFactory.spec.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { LoggerFactory } from "../Microsoft.AspNetCore.SignalR.Client.TS/Loggers"
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/LoggerFactory.spec.ts]: copyright I guess

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/IHubConnectionOptions.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { IHubProtocol } from "./IHubProtocol"
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/IHubConnectionOptions.ts]: copyright

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Loggers.ts [hunk 6]
```diff
@@ -18,3 +18,21 @@ export class ConsoleLogger implements ILogger {
         }
     }
 }
+
+export namespace LoggerFactory {
+    export function createLogger(logging?: ILogger|LogLevel) {
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Loggers.ts]: Should there be spaces around the `|`?

**Pull request merged**